src/components/Reports/StaffExpenseReport/SettingsDialog/SettingsDialog.test.tsx

```diff
@@ -32,53 +101,57 @@ describe('SettingsDialog', () => {
   });
 
   it('should render dialog when isOpen is true', () => {
-    const { getByText, getByLabelText } = render(
+    const { getByRole, getByLabelText } = render(
       <TestComponent {...defaultProps} />,
     );
 
-    expect(getByText('Report Settings')).toBeInTheDocument();
+    expect(
+      getByRole('heading', { name: 'Report Settings' }),
+    ).toBeInTheDocument();
     expect(getByLabelText('Select Date Range')).toBeInTheDocument();
-    expect(getByText('Or enter a custom date range:')).toBeInTheDocument();
-    expect(getByText('Select Categories:')).toBeInTheDocument();
+    expect(getByRole('dialog')).toBeInTheDocument();
   });
 
   it('should not render dialog when isOpen is false', () => {
-    const { queryByText } = render(
+    const { queryByRole } = render(
       <TestComponent {...defaultProps} isOpen={false} />,
     );
 
-    expect(queryByText('Report Settings')).not.toBeInTheDocument();
+    expect(queryByRole('dialog')).not.toBeInTheDocument();
   });
 
   it('should render all date range options', () => {
     const { getByRole, getByLabelText } = render(
       <TestComponent {...defaultProps} />,
     );
 
-    fireEvent.mouseDown(getByLabelText('Select Date Range'));
+    userEvent.click(getByLabelText('Select Date Range'));
 
     expect(getByRole('option', { name: 'None' })).toBeInTheDocument();
     expect(getByRole('option', { name: 'Week to Date' })).toBeInTheDocument();
     expect(getByRole('option', { name: 'Month to Date' })).toBeInTheDocument();
     expect(getByRole('option', { name: 'Year to Date' })).toBeInTheDocument();
   });
 
-  it('should render all category checkboxes', () => {
+  it('should render all category checkboxes', async () => {
     const { getByRole } = render(<TestComponent {...defaultProps} />);
 
-    expect(getByRole('checkbox', { name: 'Benefits' })).toBeInTheDocument();
+    await waitFor(() => {
+      expect(getByRole('checkbox', { name: 'Benefits' })).toBeInTheDocument();
+    });
```

Comment:
There are some other places in this file where `waitFor + getBy...` can be changed to `await findBy...`.

```suggestion
    expect(await findByRole('checkbox', { name: 'Benefits' })).toBeInTheDocument();
```